### PR TITLE
Add a prop for disabling the slider

### PR DESCRIPTION
**Pull request by reklawnos**
Fairly self-explanatory, prevents the user from moving the slider handles when `disabled=true`. I believe this covers the case where the user is moving the handle when it gets disabled, in which case it behaves like a standard HTML slider and stops in place.

### CODE REVIEW

File: src/Slider.jsx [hunk 1]
```diff
@@ -335,11 +348,15 @@ export default React.createClass({
       document.attachEvent('onmouseup', this.endSlide);
     }
 
-    return this.killEvent(ev);
```
Comment (reklawnos) [inline src/Slider.jsx]: `killEvent` doesn't return anything, so this change should have no effect.

Comment (ljharb) [inline src/Slider.jsx]: This change will violate the `consistent-return` linting rule, however.

File: src/Slider.jsx [hunk 2]
```diff
@@ -122,6 +125,12 @@ export default React.createClass({
       nextProps.orientation !== this.props.orientation
     );
 
+    const disabledChanged = nextProps.disabled !== this.props.disabled;
+
+    if (disabledChanged) {
+      this.endSlide();
```
Comment (goatslacker) [inline src/Slider.jsx]: should we place this at the bottom and short-circuit here?

**Comment by ljharb** [comment]
It seems like rather than bailing out of event handlers if the `disabled` prop is true, it would be better to simply *not apply the event handlers* in the case where it's true?

File: src/Slider.jsx (hunk 1 above)
Comment (reklawnos) [inline src/Slider.jsx]: At this point there's no other return value for `startMouseSlide`, so now this is the same as `startTouchSlide`

Comment (ljharb) [inline src/Slider.jsx]: There's a `return;` higher up in the function - any return statement means that every exit point must have a return statement.

File: src/Slider.jsx [hunk 3]
```diff
@@ -572,11 +581,13 @@ export default React.createClass({
   },
 
   render() {
+    const disabled = this.props.disabled;
```
Comment (ljharb) [inline src/Slider.jsx]: `const { disabled } = this.props;`?

Comment (reklawnos) [inline src/Slider.jsx]: While I'm at it, should I make all the other props use that style?

Comment (goatslacker) [inline src/Slider.jsx]: Sure, but can you separate this change out into its own commit?

File: src/Slider.jsx [hunk 4]
```diff
@@ -122,13 +125,19 @@ export default React.createClass({
...
     if (orientationChanged) {
       this.setState({
         className: getClassName(nextProps),
       });
     }
 
     if (minMaxChanged || valuesChanged) this.updateNewValues(nextProps);
+
+    if (disabledChanged) {
+      this.endSlide();
```
Comment (ljharb) [inline src/Slider.jsx]: maybe i'm just missing the context on `endSlide` here - but do we want it to happen whenever "disabled" changes? or just when true goes to false

Comment (reklawnos) [inline src/Slider.jsx]: Good point – just when it changes to false.

**Comment by reklawnos** [comment]
@ljharb @goatslacker Mind taking one more look at this?

File: src/Slider.jsx [hunk 5]
```diff
@@ -572,33 +581,48 @@ export default React.createClass({
...
+    const {
+      algorithm,
+      children,
+      disabled,
+      handle: Handle,
+      handleTabIndexStart,
+      max,
+      min,
+      orientation,
+      pitComponent: PitComponent,
```
Comment (ljharb) [inline src/Slider.jsx]: We should probably rename our component props to be capitalized sometime soon :-)

**Comment by ljharb** [comment]
LGTM
